ls1intum/Artemis
# `Development`: Move user management features into separate micro service

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context


This PR adds features from Artemis to the user management microservice.

### Description

This PR copies the user management REST controllers and services to the user-management microservice. Deprecated annotation is added to the copied classes and methods to make the review easier and their future removal easier. 
Only small updates (wherever it was needed) to the existing code were made. Tests are not part of this PR, otherwise, the PR would have been much larger. 
There are still some usages of the copied services in Artemis, but they will be removed in a future PR when the removal of the copied classes is done.

### Steps for Testing

At the moment this PR can only be tested locally. 
You need to run:
- JHipster Registry by running `docker-compose -f src/main/docker/jhipster-registry.yml up`
- ActiveMQ Artemis queue by running `docker-compose -f src/main/docker/activemq.yml up `
- Gateway application by running `./gradlew :gateway:bootRun`
- User Management application by running `./gradlew :user-management:bootRun --args='--spring.profiles.active=artemis,dev'`
- Artemis as usual or i.e. `./gradlew bootRun --args='--spring.profiles.active=dev,bamboo,bitbucket,artemis,scheduling' -x webapp` to skip the client code build since it's not needed

Prerequisites:
- 1 Instructor
- 1 Student

0. Open [http://localhost:8089/](http://localhost:8089/)
1. Log in to Artemis
2. Navigate to Server Administration -> User Management
3. Create, edit, remove and search for users. Each should work with no issue.
4. Navigate to any course overview page and open any of the groups: Students, Tutors, Editors, Instructors
5. Use the search bar to search for students and make sure it works
6. Enable user registration by changing application-artemis.yml
7. Try to register as a student - email won't be sent but you can make sure that the user has been created from User Management. 
8. Configure the Guided Tour tutorial as described in [here](https://docs.artemis.ase.in.tum.de/dev/guided-tour/#extend-configuration-file)
9. Create the tutorial course if not available
10. Create a new user using User Management or assign the role to an existing user 
11. Sign in with this user and check the guided tour tutorial

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Tests will be added in the next PR.






### Screenshots
